### ISSUE TITLE
Should BiologicalSequence.gap_vector return np.array?

### ISSUE THREAD

**Issue by wasade** [biocore/scikit-bio#556]
Was looking through @gregcaporaso's [alignment experiments](http://nbviewer.ipython.org/github/gregcaporaso/sketchbook/blob/master/2014.07.12-alignment-experiments/alignment-filtering.ipynb) and saw that `gap_vector` was returning a `list`. Since this is a boolean vector, I wanted to check if there would be support in having this cast to `np.array`? In the context of the alignment experiments, this would (with the current versions of numpy) likely suggest using `searchsorted` to find the start and end positions instead of `index`. The two direct benefits are that `searchsorted` is O(log N) vs. O(N) for `index`, and second that this then allows manipulation of the `gap_vector` with numpy logical operations.

The specific code can be found [here](
https://github.com/biocore/scikit-bio/blob/master/skbio/core/sequence.py#L803), with the change being to cast to `np.array`.

**Comment by jairideout** [comment]
Fixed in #920:

```python
In [3]: DNA('AC--G.T').gaps()
Out[3]: array([False, False,  True,  True, False,  True, False], dtype=bool)
```

**Comment by ebolyen** [comment]
:+1: This is so cool because it can be composed with the rest of the Sequence API now as well!
